cloudflare_rules/cloudflare_firewall_suspicious_event_greynoise.py

```diff
@@ -0,0 +1,37 @@
+from ipaddress import ip_address
+
+from panther_greynoise_helpers import GetGreyNoiseObject, GetGreyNoiseRiotObject
+
+
+def rule(event):
+    if event.get("Action") == "block":
+        return False
+
+    # Validate the IP is actually an IP
+    try:
+        ip_address(event.get("ClientIP"))
+    except ValueError:
+        return False
+
+    # Setup GreyNoise variables
+    global NOISE  # pylint: disable=global-variable-undefined
+    NOISE = GetGreyNoiseObject(event)
+    riot = GetGreyNoiseRiotObject(event)
+
+    # If IP is in the RIOT dataset, we can assume it is safe
+    if riot.is_riot("ClientIP"):
+        return False
+
+    # Check if IP classified as malicious
+    if NOISE.classification("ClientIP") == "malicious":
+        return True
+
+    return False
+
+
+def title(event):
+    return f"Suspicious Event Detected - <{event.get('ClientIP', 'UNKNOWN_IP')}>"
```

Comment:
I think this could be a bit more descriptive, maybe something like "Request allowed from malicious IP" or something like that?

Reply:
Removed the conditional that checks on whether or not the request was blocked and left the title the same (relying heavily on the GreyNoise dataset to determine malicious intent)

Reply:
Is this now equivalent to `cloudflare_rules/cloudflare_firewall_high_volume_events_greynoise.py` though? Just without the threshold?

Reply:
ah nvm I see this is allowed & blocked connections where ^^ is just blocked. I feel like there is value in alerting on only allowed connections though right? If it is blocked, I probably don't care a whole lot unless the volume is really high or abnormal (which is exactly what you built above)

Reply:
What am I missing? I don't see a conditional in this rule that would return false if the action was blocked

Reply:
Okay yeah, good catch -- I'll add a conditional to the suspicious events rule to only alert on non-blocked events

Reply:
Doh, okay yeah that conditional should have been left in for this one to only alert on non-blocked events